src/circuits/etable_compact/op_configure/mod.rs

```diff
@@ -340,6 +386,15 @@ impl<'a, F: FieldExt> EventTableCellAllocator<'a, F> {
         }
     }
 
+    pub fn alloc_u4_bop(&mut self) -> U4BopCell {
+        assert!(self.u4_bop_index < 1 as i32);
+        let allocated_index = self.u4_bop_index;
```

Comment:
looks like an unused variable, please remove it.

Reply:
The interface works, but I prefer to return the entire 16 lines of (u4 bop, u4 lhs, u4 rhs, u4 res) in one allocation to avoid misuse. So it may be changed in the future.

Reply:
> looks like an unused variable, please remove it.

removed, thanks!

Reply:
> The interface works, but I prefer to return the entire 16 lines of (u4 bop, u4 lhs, u4 rhs, u4 res) in one allocation to avoid misuse. So it may be changed in the future.

I think it makes sense to return 16 lines of u4_bop in one allocation. The u4 lhs etc are allocated by alloc_u64 etc. Let me know what do you think. If it sounds good, I will change it to return 16 of u4_bop cells.

Reply:
Then I think your code is good enough. Let's put off the change of 16 lines allocation to a scenario that needs to alloc u4_bop twice in a op_class.